app/src/main/java/org/mozilla/focus/widget/FloatingActionButtonBehavior.java

```diff
@@ -74,9 +74,15 @@ private void hideButton() {
         animate(button, true);
     }
 
-    private void animate(View child, final boolean hide) {
+    private void animate(final View child, final boolean hide) {
         animating = true;
 
+        if (!hide) {
+            // Ensure the child will be visible before starting animation: if it's hidden, we've also
+            // set it to View.GONE, so we need to restore that now, _before_ the animation starts.
+            child.setVisibility(View.VISIBLE);
```

Comment:
Instead of doing it here you can override onAnimationStart() in the AnimatorListenerAdapter below.